pkg/scalers/aws_sqs_queue_scaler.go

```diff
@@ -115,6 +115,8 @@ func parseAwsSqsQueueMetadata(config *ScalerConfig, logger logr.Logger) (*awsSqs
 
 	if val, ok := config.TriggerMetadata["queueURL"]; ok && val != "" {
 		meta.queueURL = val
+	} else if val, ok := config.TriggerMetadata["queueURLFromEnv"]; ok && val != "" {
```

Comment:
The question is, should you introduce this property, or allow trigger authentication to specify queueUrl instead?

Reply:
I reused the authentication map as it was being passed to resolvedEnv. I think we'd want it as a property separate from trigger auth. I just made an update to the test case to reflect that.
https://github.com/kedacore/keda/blob/708890964f00c169ba990fa5fd497b47893a4ded/pkg/scalers/aws_sqs_queue_scaler_test.go#L258

Reply:
No, what Tom is refering to here is. There is already a way to set a parameter in `ScaledObject` metadata using environment variables without an explicit `parameterFromEnv` parameter in the metadata. 

In the `TriggerAuth`, pull the environment variable (refer [this](https://keda.sh/docs/2.8/authentication-providers/environment-variable/)), and then set the `TriggerAuth` as the `authenticationRef` in the `ScaledObject`.

Reply:
So the question he is asking is should the change introduce a `queueURLFromEnv` parameter and pull it from `ResolvedEnv` or use the `TriggerAuth` and pull from `AuthParams`

Reply:
For example, the EventHub scaler supports pulling from both.
https://github.com/kedacore/keda/blob/db6ba503ec8dcde76d6aa1e7979857df18dff732/pkg/scalers/azure_eventhub_scaler.go#L118-L122

Reply:
Oh I see. Given that some of the other scalers have `*FromEnv`, I think it might be good to also introduce it here, unless there's a plan to deprecate them in the future in favour of setting them in the `AuthParams`.
What do you prefer in this case? Should it be done to support both or just loading from `AuthParams`?

Reply:
@tomkerkhove Are there any plans to remove *FromEnv in the long run and only have the auth param values? Or should this PR just implement both?

Reply:
@v-shenoy Any thoughts on this?

Reply:
This is up to the maintainers @tomkerkhove @JorTurFer @zroubalik.

Reply:
I'm not sure about this, but I think that when I started with KEDA I heard that we want to avoid `*FromEnv` because TriggerAuthentication is the way and it already supports getting values from environment. 
IMO, I prefer to unify every env access under TriggerAuthentication instead of getting values from environment using 2 different ways, but it's true oldest scalers supports both.
If I had been who is doing this feature, I would support only TriggerAuthentication as a source for getting the value from environment but let's see other opinions

Reply:
I am not opposed having `*FromEnv` where it makes sense.

Reply:
just to clarify, me neither xD

Reply:
Should I then make the change to support both `*FromEnv` and TriggerAuthentication? So far I've only made changes to support `*FromEnv`

Reply:
Let's wait other opinions but from my side, we can proceed as it is